Assets/MixedRealityToolkit/Inspectors/Profiles/BaseMixedRealityProfileInspector.cs

```diff
@@ -132,76 +130,13 @@ protected static bool RenderProfile(SerializedProperty property, Type profileTyp
-                        {
-                            Debug.Assert(profileTypeName != null, "No Type Found");
-
-                            ScriptableObject instance = CreateInstance(profileTypeName);
-                            var newProfile = instance.CreateAsset(AssetDatabase.GetAssetPath(Selection.activeObject)) as BaseMixedRealityProfile;
-                            property.objectReferenceValue = newProfile;
-                            property.serializedObject.ApplyModifiedProperties();
-                            changed = true;
-                        }
-                    }
-                }
-                else
-                {
-                    var renderedProfile = property.objectReferenceValue as BaseMixedRealityProfile;
-                    Debug.Assert(renderedProfile != null);
-                    Debug.Assert(profile != null, "No profile was set in OnEnable. Did you forget to call base.OnEnable in a derived profile class?");
-
-                    if (GUILayout.Button(new GUIContent("Clone", "Replace with a copy of the default profile."), EditorStyles.miniButton, GUILayout.Width(42f)))
-                    {
-                        MixedRealityProfileCloneWindow.OpenWindow(profile, renderedProfile, property);
-                    }
-                }
-            }
+            // Draw the profile dropdown
+            changed |= MixedRealityInspectorUtility.DrawProfileDropDownList(property, profile, oldObject, profileType, showAddButton);
 
-            if (property.objectReferenceValue != null)
-            {
-                UnityEditor.Editor subProfileEditor = UnityEditor.Editor.CreateEditor(property.objectReferenceValue);
-
-                // If this is a default MRTK configuration profile, ask it to render as a sub-profile
-                if (typeof(BaseMixedRealityToolkitConfigurationProfileInspector).IsAssignableFrom(subProfileEditor.GetType()))
-                {
-                    BaseMixedRealityToolkitConfigurationProfileInspector configProfile = (BaseMixedRealityToolkitConfigurationProfileInspector)subProfileEditor;
-                    configProfile.RenderAsSubProfile = true;
-                }
-
-                var subProfile = property.objectReferenceValue as BaseMixedRealityProfile;
-                if (subProfile != null && !subProfile.IsCustomProfile)
-                {
-                    EditorGUILayout.HelpBox("Clone this default profile to edit properties below", MessageType.Warning); 
-                }
-
-                if (renderProfileInBox)
-                {
-                    EditorGUILayout.BeginVertical(EditorStyles.helpBox);
-                }
-                else
-                {
-                    EditorGUILayout.BeginVertical();
-                }
+            Debug.Assert(profile != null, "No profile was set in OnEnable. Did you forget to call base.OnEnable in a derived profile class?");
```

Comment:
Why would profile be set in OnEnable? Not understanding the assert comment?

Also should this render a warning/error in the inspector? Because this will likely spam the console no? On every repaint

Reply:
It was a check in the original base class. I've never actually seen this assert fail in the wild. I'm going to remove it.